### PR TITLE
DIV-6499-Extend-multiple-events-so-ServiceApplicationNotApproved-is-part-of-pre-state-events

### PR DESCRIPTION
**Pull request by madjava**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/DIV-6499

### Change description ###
Extending the ways, state that would enable the process to continue when a case is in the ServiceApplicationNotApproved state also adding more role can see this event.

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: definitions/divorce/json/CaseEvent/CaseEvent-nonprod.json
Comment (madjava) [inline definitions/divorce/json/CaseEvent/CaseEvent-nonprod.json:39]: As per AC's for this ticket `allowPetitionerToAmend` and `amendPetitionForRefusalSol` pre-state are to be extended to include `ServiceApplicationNotApproved`.  Issue i faced was that these events are part of the `pet-amend` epic and the new pre-state is part of `deemed-and-dispensed` epic. I had originally separated them in respective feature files but CCD build was failing as it saw duplicate events in the definition. My solution was to move them to the `nonprod` file. Seeing that the service application is part of another event, if the `pet-amentd` event gets released first the the `ServiceApplicationNotApproved` in this definition would go also. This might not be an issue as the `RBAC` for this state are in the right feature files i.e. the `deemed-and-dispensed` one, just though i'd mention this here explain why this definition is in this file and not either `pet-amend` or `deemed-and-dispensed` feature files and to draw your attention to it. @yarpo @Dizrupto please advise.

File: definitions/divorce/json/CaseEvent/CaseEvent-pet-amend-nonprod.json
Comment (madjava) [inline definitions/divorce/json/CaseEvent/CaseEvent-pet-amend-nonprod.json:14]: Moved to `-nonprod` file

File: definitions/divorce/json/CaseEvent/CaseEvent-pet-amend-nonprod.json
Comment (madjava) [inline definitions/divorce/json/CaseEvent/CaseEvent-pet-amend-nonprod.json:37]: Moved to '-nonprod` file

File: definitions/divorce/json/CaseEvent/CaseEvent-prod.json
Comment (madjava) [inline definitions/divorce/json/CaseEvent/CaseEvent-prod.json:38]: Reformatting, definition below

File: definitions/divorce/json/CaseEvent/CaseEvent.json
Comment (madjava) [inline definitions/divorce/json/CaseEvent/CaseEvent.json:230]: Moved to `-prod` file

File: definitions/divorce/json/CaseEvent/CaseEvent.json
Comment (madjava) [inline definitions/divorce/json/CaseEvent/CaseEvent.json:2501]: Moved to `-prod` file

File: definitions/divorce/json/CaseEvent/CaseEvent.json
Comment (madjava) [inline definitions/divorce/json/CaseEvent/CaseEvent.json:1158]: Moved to `-prod` file

File: definitions/divorce/json/CaseEvent/CaseEvent-deemed-and-dispensed-nonprod.json
Comment (madjava) [inline definitions/divorce/json/CaseEvent/CaseEvent-deemed-and-dispensed-nonprod.json:118]: Extending required pre-states to include `ServiceApplicationNotApproved`

File: definitions/divorce/json/AuthorisationCaseState/AuthorisationCaseState-deemed-and-dispensed-nonprod.json
Comment (madjava) [inline definitions/divorce/json/AuthorisationCaseState/AuthorisationCaseState-deemed-and-dispensed-nonprod.json:148]: Updating `ServiceApplicationNotApproved` RBAC for other roles

File: definitions/divorce/json/CaseEvent/CaseEvent-deemed-and-dispensed-nonprod.json
Comment (yarpo) [inline definitions/divorce/json/CaseEvent/CaseEvent-deemed-and-dispensed-nonprod.json:59]: It doesn't matter for CCD, but it's quite useful to use current date here.

File: definitions/divorce/json/CaseEvent/CaseEvent-deemed-and-dispensed-nonprod.json
Comment (yarpo) [inline definitions/divorce/json/CaseEvent/CaseEvent-deemed-and-dispensed-nonprod.json:74]: as above

File: definitions/divorce/json/CaseEvent/CaseEvent-deemed-and-dispensed-nonprod.json
Comment (yarpo) [inline definitions/divorce/json/CaseEvent/CaseEvent-deemed-and-dispensed-nonprod.json:89]: as above

File: definitions/divorce/json/CaseEvent/CaseEvent-deemed-and-dispensed-nonprod.json
Comment (yarpo) [inline definitions/divorce/json/CaseEvent/CaseEvent-deemed-and-dispensed-nonprod.json:104]: the same

File: definitions/divorce/json/CaseEvent/CaseEvent-nonprod.json
Comment (yarpo) [inline definitions/divorce/json/CaseEvent/CaseEvent-nonprod.json:14]: same

File: definitions/divorce/json/CaseEvent/CaseEvent-nonprod.json
Comment (yarpo) [reply definitions/divorce/json/CaseEvent/CaseEvent-nonprod.json:39]: You need to wait until I merge this: https://github.com/hmcts/div-ccd-definitions/pull/483

Nothing else we can do.

File: definitions/divorce/json/CaseEvent/CaseEvent-pet-amend-nonprod.json
Comment (yarpo) [reply definitions/divorce/json/CaseEvent/CaseEvent-pet-amend-nonprod.json:14]: it will not help.

File: definitions/divorce/json/CaseEvent/CaseEvent-nonprod.json
Comment (Dizrupto) [reply definitions/divorce/json/CaseEvent/CaseEvent-nonprod.json:39]: Yep that's a good point. I guess when Patryk's PR is merged you can move it to the 'deemed-and-dispensed' non-prod file so we make sure to release it at the same time.

**Review by Dizrupto — APPROVED** [review]

File: definitions/divorce/json/CaseEvent/CaseEvent-deemed-and-dispensed-nonprod.json
Comment (madjava) [reply definitions/divorce/json/CaseEvent/CaseEvent-deemed-and-dispensed-nonprod.json]: Updated

File: definitions/divorce/json/CaseEvent/CaseEvent-deemed-and-dispensed-nonprod.json
Comment (madjava) [reply definitions/divorce/json/CaseEvent/CaseEvent-deemed-and-dispensed-nonprod.json]: Updated

File: definitions/divorce/json/CaseEvent/CaseEvent-deemed-and-dispensed-nonprod.json
Comment (madjava) [reply definitions/divorce/json/CaseEvent/CaseEvent-deemed-and-dispensed-nonprod.json]: Updated

File: definitions/divorce/json/CaseEvent/CaseEvent-deemed-and-dispensed-nonprod.json
Comment (madjava) [reply definitions/divorce/json/CaseEvent/CaseEvent-deemed-and-dispensed-nonprod.json]: Updated

File: definitions/divorce/json/CaseEvent/CaseEvent-nonprod.json
Comment (madjava) [reply definitions/divorce/json/CaseEvent/CaseEvent-nonprod.json]: Updated

File: definitions/divorce/json/CaseEvent/CaseEvent-nonprod.json
Comment (madjava) [reply definitions/divorce/json/CaseEvent/CaseEvent-nonprod.json:39]: Ok, thanks @Dizrupto

File: definitions/divorce/json/CaseEvent/CaseEvent-pet-amend-nonprod.json
Comment (yarpo) [reply definitions/divorce/json/CaseEvent/CaseEvent-pet-amend-nonprod.json:37]: nonprod was meant for different scenarios (things that should never be deployed - like "testEventToMoveCaseToGivenState")

File: definitions/divorce/json/CaseEvent/CaseEvent-prod.json
Comment (yarpo) [inline definitions/divorce/json/CaseEvent/CaseEvent-prod.json:3]: let's have a call. I don't really get it.

Is your story a part of any release? If not - you can't update prod files. It will change prod xlsx -> which means this should be re-uploaded to prod.

Re-uploading prod xlsx to prod === release.

Comment (yarpo) [reply definitions/divorce/json/CaseEvent/CaseEvent-prod.json:3]: unless I'm wrong and this will not change anything in a prod xlsx :)

Comment (madjava) [reply definitions/divorce/json/CaseEvent/CaseEvent-prod.json:3]: Ok, that should not have been. Must have been an error on my part. Prod should not have been touch. Let me verify. Thanks

**Review by yarpo — APPROVED** [review]

**Review by yarpo — APPROVED** [review]

**Pull request merged**
